### PR TITLE
Add participating stores to Rewards screen

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Added the list of stores which accept rewards to the 'Healthy Rewards' tab.
- Clicking the store name redirects to the map (just as clicking the store card in `StoreListScreen` does)

### Unrelated fixes
- Pulled `rewardPointValue` and `rewardDollarValue` from `/constants` for the cards from #93 
- Added `key` for those lists (would happen rarely in PROD assuming customers have small total purchase amounts in $)
- Added testing bypass button back in `dev` mode (necessary to test this feature since you need to be able to see rewards)
- Renamed a function because it was no longer transitioning to `StoreDetails` LOL

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources

We should fix this error probably? "never" is a scary word 🤷‍♀️ (lol)
https://www.lahteenlahti.com/fixing-virtualizedlists-should-never-be-nested-inside-plain-scrollviews/

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

branches off #128 
update #93 

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

### Styling
Not sure if spacing still needs to be adjusted, especially the indent.

### Functionality
Log in, make sure nothing breaks, and that stores redirect correctly.

Try updating the `rewards accepted` value in the [DEV] Airtable **(but remember to switch it back after, please)**

If erroneous lint/prop-types errors show up, please review #128 

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

N/A 

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

- This won't scale well (i.e when > 5 stores are accepting rewards the design should probably be updated, or this feature removed entirely).

* Noticed some very edge-case bugs (should _never_ happen in PROD, though) unrelated to the current PR
* If transaction does not have a `rewardsUnlocked` or `rewardsApplied` value (or whichever) the `Array(rewardsUnlocked)...` throws an error like 'Array size too big', probably because the value is undefined

### Screenshots

image

https://www.loom.com/share/8f9166e49f9544edae54188634e6bd53

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/rewards/RewardsHome.js [hunk 1]
```diff
@@ -25,11 +27,31 @@ function createList(n) {
...
+        <Overline style={{ marginTop: 24, marginBottom: 4 }}>
+          Participating Stores
+        </Overline>
+        {participating.map((store) => {
+          return (
+            <Subhead
+              style={{ marginLeft: 8 }}
+              key={store.id}
+              onPress={() =>
+                navigation.navigate('Stores', {
```
Comment (wangannie) [inline components/rewards/RewardsHome.js:46]: Hmm this is really really slow to load for some reason. Could we do first do a `navigation.goBack` then do `changeCurrentStore` from MapScreen to set the store if that's not too clunky?

Comment (wangannie) [reply components/rewards/RewardsHome.js:46]: Just so it doesn't leave the user staring at the screen wondering what's happening/what's supposed to happen.

Also, can you wrap these in touchable opacities so there's some response when these are clicked? Even that might help indicate that some action is supposed to happen bc rn it's quite ambiguous.

Comment (anniero98) [reply components/rewards/RewardsHome.js:46]: > Could we do first do a navigation.goBack then do changeCurrentStore from MapScreen to set the store if that's not too clunky?

AFAIK this isn't possible? The method that gets invoked is `componentWillReceiveProps`, which is setting the store in `MapScreen`

> can you wrap these in touchable opacities so there's some response when these are clicked

👍

File: components/rewards/RewardsHome.js [hunk 2]
```diff
@@ -25,11 +34,32 @@ function createList(n) {
...
 
-function RewardsHome({ customer }) {
+function RewardsHome({ customer, participating }) {
+  const navigation = useNavigation();
   const rewardsAvailable = parseInt(customer.points, 10) / rewardPointValue;
   const pointsToNext = parseInt(customer.points, 10) % rewardPointValue;
   return (
     <ScrollView style={{ marginLeft: 16, paddingRight: 16 }}>
+      <ColumnContainer>
+        <Overline style={{ marginTop: 24, marginBottom: 4 }}>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js:44]: See the latest designs -- this is moved to the bottom (under Avail rewards).

Also, this is included in the rewards screen even in Guest Mode

**Comment by anniero98** [comment]
Updated styling + added to guest mode
image
image

File: components/rewards/ParticipatingStores.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  return (
+    <ColumnContainer
+      style={guest ? { marginLeft: 16, marginBottom: 24 } : { marginTop: 28 }}>
+      <Overline style={{}}>Participating Stores</Overline>
+      {participating.map((store) => {
+        return (
+          <TouchableOpacity key={store.id}>
+            <Subhead
+              style={{ marginLeft: 12 }}
+              onPress={() =>
```
Comment (wangannie) [inline components/rewards/ParticipatingStores.js:19]: When I click this, it doesn't show the map marker as selected.
